src/bin/login/shard.rs

```diff
@@ -550,3 +551,79 @@ pub fn buddy_menuchat_succ(
 
     Ok(())
 }
+
+pub fn buddy_warp(
+    shard_key: usize,
+    clients: &mut HashMap<usize, FFClient>,
+    state: &mut LoginServerState,
+) -> FFResult<()> {
+    let server = clients.get_mut(&shard_key).unwrap();
+    let pkt: sP_FE2LS_REQ_BUDDY_WARP = *server.get_packet(P_FE2LS_REQ_BUDDY_WARP)?;
+
+    let buddy_uid = pkt.iBuddyPCUID;
+    let pc_id = pkt.iPC_ID;
+
+    let buddy_shard_id = match state.get_player_shard(buddy_uid) {
+        Some(shard_id) => shard_id,
+        None => {
+            log(
+                Severity::Warning,
+                &format!(
+                    "Buddy warp request: buddy UID {} not found in any shard",
+                    buddy_uid
+                ),
+            );
+            return Ok(());
+        }
+    };
+
+    let buddy_metadata = match state.get_player_metadata(buddy_uid) {
+        Some(metadata) => metadata,
+        None => {
+            log(
+                Severity::Warning,
+                &format!(
+                    "Buddy warp request: buddy UID {} metadata not found",
+                    buddy_uid
+                ),
+            );
+            return Ok(());
+        }
+    };
+
+    if buddy_metadata.map_num != 0 {
+        log(
+            Severity::Warning,
```

Comment:
this can probably be less important
```suggestion
            Severity::Info,
```